code/game/machinery/vending/cm_vending.dm

```diff
@@ -902,108 +873,180 @@ IN_USE						used for vending/denying
 	. = ..()
 	populate_product_list(1.2)
 
+	for (var/i in 1 to length(listed_products))
+		// initial item count setup
+		var/item_name = listed_products[i][1]
+		var/initial_count = listed_products[i][2]
+		initial_product_count[item_name] = initial_count
+
+		// icon setup
+		var/typepath = listed_products[i][3]
+		var/obj/item/I = typepath
```

Comment:
![image](https://user-images.githubusercontent.com/66756236/190900680-2bf23a5e-5866-4d8c-a838-c8e020b507f8.png)
issues caused when the listed product is an /obj/effect not an /item
i.e. here
![image](https://user-images.githubusercontent.com/66756236/190900704-0c8072f1-1314-4c56-b980-bdb0b60a131b.png)

Reply:
Thanks for the feedback - I'm starting to feel that this was quite a bit more in depth than I originally thought. Will try and resolve these issues today.

Reply:
Managed to fix the icon issues, lots of little things adding up